drivers/serial/uart_infineon_cat1.c

```diff
@@ -0,0 +1,506 @@
+/* Copyright 2022 Cypress Semiconductor Corporation (an Infineon company) or
+ * an affiliate of Cypress Semiconductor Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @brief UART driver for Infineon CAT1 MCU family.
+ *
+ * Note:
+ * - Uart ASYNC functionality is not implemented in current
+ *   version of Uart CAT1 driver.
+ */
+
+#define DT_DRV_COMPAT infineon_mtbhal_uart
+
+#include <device.h>
+#include <drivers/uart.h>
+#include <drivers/pinctrl.h>
+#include "cyhal_uart.h"
```

Comment:
Please use `<...>` as this is not a local include.

Reply:
updated